### PR TITLE
6670 - Meta's end field not refreshing when saving contact form

### PR DESCRIPTION
**Pull request by latin-panda**
# Description

This PR:
- Adds a trigger of Enketo beforeSave event when saving contact form, this will refresh the meta's fields (Reports already has it)
- Filters out the 1st level fields in contact form since it only process groups to be docs, this prevents incomplete data in couch's docs when saving the place. I found this behaviour while testing the meta fields. 

https://github.com/medic/cht-core/issues/6670

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: webapp/src/ts/services/enketo-translation.service.ts [hunk 1]
```diff
@@ -159,28 +159,30 @@ export class EnketoTranslationService {
...
       result.repeats = repeats;
     }
 
     const NODE_NAMES_TO_IGNORE = ['meta', 'inputs', 'repeat'];
 
-    this.withElements(root.childNodes)
-      .filter((node:any) => !NODE_NAMES_TO_IGNORE.includes(node.nodeName))
+    this
+      .withElements(root.childNodes)
+      .filter((node:any) => !NODE_NAMES_TO_IGNORE.includes(node.nodeName) && node.childElementCount > 0)
```
Comment (latin-panda) [inline webapp/src/ts/services/enketo-translation.service.ts:176]: This filters out 1st level fields and empty groups to only process groups with content to be docs.

**Comment by latin-panda** [comment]
Hi @jkuester, when you have time, can you please review? Thanks!

**Review by jkuester — APPROVED** [review]
Looks good!   Just wanted to add for the record (in case someone wonders later) that the reason for filtering nodes that have no child elements in the translation service is that the top level elements on contact forms [have very specific functionality and constraints](https://docs.communityhealthtoolkit.org/apps/guides/forms/contact-forms/) (when it comes to storing data back to the database.  Only the top-level elements that are groups should be saved.

File: webapp/src/ts/services/enketo-translation.service.ts (hunk 1 above)
Comment (jkuester) [reply webapp/src/ts/services/enketo-translation.service.ts:176]: @latin-panda would it be a good idea to have a warning in `cht-conf` if someone tries to compile a contact form with 1st level fields that are not groups?

Comment (latin-panda) [reply webapp/src/ts/services/enketo-translation.service.ts:176]: 🤔 I wonder if it's going to be helpful or annoying, depending how frequently app developers use top level fields for calculations inside groups vs how often this turns out to be a mistake/error in their forms.

Comment (jkuester) [reply webapp/src/ts/services/enketo-translation.service.ts:176]: @latin-panda good point. I forgot that it is probably pretty common to just put calcs up at the top that might get referenced other places throughout the form.  Out of curiosity, what is happening to those calc values when the form is saved right now (before these changes)?  (I get from the described issue that they are not properly saved anywhere, but are they just dropped or are they thrown into some unreferenced doc?)

**Pull request merged**